diff --git a/tests/core/adapter/BUILD.gn b/tests/core/adapter/BUILD.gn
index 80a043783..2d6bd0c7b 100644
--- a/tests/core/adapter/BUILD.gn
+++ b/tests/core/adapter/BUILD.gn
@@ -254,8 +254,8 @@ ohos_unittest("LNNOhosAccountMockTest") {
   ]
   if (is_standard_system) {
     external_deps += [
-      "os_account:libaccountkits",
-      "os_account:os_account_innerkits",
+      #"os_account:libaccountkits",
+      #"os_account:os_account_innerkits",
     ]
   }
 }
@@ -303,8 +303,8 @@ ohos_unittest("LnnOhosAccountAdapterTest") {
         "googletest:gtest_main",
         "hilog:libhilog",
         "ipc:ipc_single",
-        "os_account:libaccountkits",
-        "os_account:os_account_innerkits",
+        #"os_account:libaccountkits",
+        #"os_account:os_account_innerkits",
       ]
     }
   } else {
@@ -407,7 +407,7 @@ ohos_unittest("WrapperBrInterfaceTest") {
   ]
 
   external_deps = [
-    "bluetooth:btframework",
+    #"bluetooth:btframework",
     "c_utils:utils",
     "googletest:gmock_main",
     "hilog:libhilog",
@@ -420,18 +420,26 @@ group("unittest") {
   deps = [
     ":AdapterDsoftbusRsaCryptoTest",
     ":LNNKvStoreLaunchListenerTest",
-    ":LNNOhosAccountMockTest",
-    ":LNNOhosAccountTest",
     ":LnnSaStatusMonitorTest",
-    ":WrapperBrInterfaceTest",
   ]
+  if (support_jsapi) {
+    deps += [ ":WrapperBrInterfaceTest" ]
+  }
+  if (support_jsapi) {
+    deps += [
+      ":LNNOhosAccountMockTest",
+      ":LNNOhosAccountTest",
+    ]
+  }
   if (dsoftbus_feature_lnn_cloud_sync) {
     deps += [
       ":LnnKVAdapterTest",
       ":LnnKVAdapterWrapperTest",
-      ":LnnOhosAccountAdapterTest",
       ":LnnSettingdataEventMonitorTest",
     ]
+    if (support_jsapi) {
+      deps += [ ":LnnOhosAccountAdapterTest" ]
+    }
   }
   if (dsoftbus_feature_lnn_usb_ncm && dsoftbus_feature_deps_usb_manager) {
     deps += [ ":LnnUsbAdapterTest" ]
diff --git a/tests/core/authentication/BUILD.gn b/tests/core/authentication/BUILD.gn
index 1cb58fe6b..afd8343ad 100644
--- a/tests/core/authentication/BUILD.gn
+++ b/tests/core/authentication/BUILD.gn
@@ -852,8 +852,8 @@ ohos_unittest("AuthDeviceProfileTest") {
       "googletest:gtest_main",
       "hilog:libhilog",
       "ipc:ipc_core",
-      "os_account:libaccountkits",
-      "os_account:os_account_innerkits",
+      #"os_account:libaccountkits",
+      #"os_account:os_account_innerkits",
       "samgr:samgr_proxy",
     ]
   } else {
@@ -865,8 +865,8 @@ ohos_unittest("AuthDeviceProfileTest") {
       "googletest:gtest_main",
       "hilog:libhilog",
       "ipc:ipc_core",
-      "os_account:libaccountkits",
-      "os_account:os_account_innerkits",
+      #"os_account:libaccountkits",
+      #"os_account:os_account_innerkits",
     ]
   }
 }
@@ -1983,7 +1983,6 @@ group("unittest") {
     ":AuthApplyKeyManagerTest",
     ":AuthApplyKeyProcessTest",
     ":AuthConnectionTest",
-    ":AuthDeviceProfileTest",
     ":AuthEnhanceMockTest",
     ":AuthHichainIdServiceTest",
     ":AuthHichainMockTest",
@@ -2008,9 +2007,14 @@ group("unittest") {
     ":AuthUserCommonKeyNewTest",
     ":AuthUserCommonKeyTest",
   ]
+  if (support_jsapi) {
+    deps += [ ":AuthDeviceProfileTest" ]
+  }
   if (!defined(global_parts_info) ||
       defined(global_parts_info.deviceprofile_device_info_manager)) {
-    deps += [ ":AuthDeviceProfileListenerTest" ]
+    if (support_jsapi) {
+      deps += [ ":AuthDeviceProfileListenerTest" ]
+    }
   }
 }
 
diff --git a/tests/core/bus_center/lnn/BUILD.gn b/tests/core/bus_center/lnn/BUILD.gn
index a0e63d237..e6e052f6a 100644
--- a/tests/core/bus_center/lnn/BUILD.gn
+++ b/tests/core/bus_center/lnn/BUILD.gn
@@ -2416,7 +2416,7 @@ ohos_unittest("LNNHeartBeatTest") {
   ]
 
   external_deps += [
-    "bluetooth:btframework",
+    #"bluetooth:btframework",
     "googletest:gmock",
     "googletest:gtest_main",
     "hilog:libhilog",
@@ -2506,7 +2506,7 @@ ohos_unittest("LnnHeartBeatDeviceRiskTest") {
   ]
 
   external_deps += [
-    "bluetooth:btframework",
+    #"bluetooth:btframework",
     "googletest:gmock",
     "googletest:gtest_main",
     "hilog:libhilog",
@@ -2596,7 +2596,7 @@ ohos_unittest("LnnHeartBeatConstraintTest") {
   ]
 
   external_deps += [
-    "bluetooth:btframework",
+    #"bluetooth:btframework",
     "googletest:gmock",
     "googletest:gtest_main",
     "hilog:libhilog",
diff --git a/tests/core/common/security/permission/standard/BUILD.gn b/tests/core/common/security/permission/standard/BUILD.gn
index b3cb75abe..840c887ea 100644
--- a/tests/core/common/security/permission/standard/BUILD.gn
+++ b/tests/core/common/security/permission/standard/BUILD.gn
@@ -19,6 +19,9 @@ module_output_path = "dsoftbus/soft_bus/transmission"
 ohos_unittest("SoftbusPermissionTest") {
   module_out_path = module_output_path
   sources = [ "unittest/softbus_permission_test.cpp" ]
+  if (support_jsapi) {
+    defines = [ "SUPPORT_JSAPI" ]
+  }
 
   include_dirs = [
     "$dsoftbus_root_path/core/common/include",
diff --git a/tests/core/common/security/permission/standard/unittest/softbus_permission_test.cpp b/tests/core/common/security/permission/standard/unittest/softbus_permission_test.cpp
index b5cd7b94c..7e475355b 100644
--- a/tests/core/common/security/permission/standard/unittest/softbus_permission_test.cpp
+++ b/tests/core/common/security/permission/standard/unittest/softbus_permission_test.cpp
@@ -395,6 +395,7 @@ HWTEST_F(SoftbusPermissionTest, PermStateChangeCallback003, TestSize.Level0)
     std::cout << "g_btPermissionChangeCb is empty." << std::endl;
 }
 
+#ifdef SUPPORT_JSAPI
 /**
  * @tc.name:PermStateChangeCallback004
  * @tc.desc: Test PermStateChangeCallback when vilide param
@@ -415,6 +416,7 @@ HWTEST_F(SoftbusPermissionTest, PermStateChangeCallback004, TestSize.Level0)
     EXPECT_NO_FATAL_FAILURE(accessTokenAdapterObj.PermStateChangeCallback(result));
     std::cout << "g_btPermissionChangeCb is not empty." << std::endl;
 }
+#endif
 
 /*
  * @tc.name:SoftBusRegisterPermission001
@@ -587,4 +589,4 @@ HWTEST_F(SoftbusPermissionTest, SoftBusGetNativeProcessNameTest002, TestSize.Lev
     int32_t ret = SoftBusGetNativeProcessName(tokenId, nullptr, PROCESS_NAME_SIZE_MAX);
     EXPECT_EQ(ret, SOFTBUS_INVALID_PARAM);
 }
-} // namespace OHOS
\ No newline at end of file
+} // namespace OHOS
diff --git a/tests/core/connection/fuzztest/conngeneralconnection_fuzzer/BUILD.gn b/tests/core/connection/fuzztest/conngeneralconnection_fuzzer/BUILD.gn
index 0cedf6998..d376db8b2 100644
--- a/tests/core/connection/fuzztest/conngeneralconnection_fuzzer/BUILD.gn
+++ b/tests/core/connection/fuzztest/conngeneralconnection_fuzzer/BUILD.gn
@@ -84,14 +84,14 @@ ohos_fuzztest("ConnGeneralConnectionFuzzTest") {
     "$dsoftbus_root_path/dfx:softbus_dfx",
   ]
 
-  if (dsoftbus_feature_conn_br) {
+  if (dsoftbus_feature_conn_br && support_jsapi) {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common.c" ]
   } else {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common_virtual.c" ]
   }
 
   external_deps = [
-    "bluetooth:btframework",
+    #"bluetooth:btframework",
     "bounds_checking_function:libsec_shared",
     "cJSON:cjson",
     "c_utils:utils",
diff --git a/tests/core/connection/general/BUILD.gn b/tests/core/connection/general/BUILD.gn
index f07c43fa7..3703afe90 100644
--- a/tests/core/connection/general/BUILD.gn
+++ b/tests/core/connection/general/BUILD.gn
@@ -72,7 +72,7 @@ ohos_unittest("GeneralConnectionTest") {
     "$dsoftbus_root_path/tests/core/connection/general/unittest/softbus_conn_general_connection_test.cpp",
   ]
 
-  if (dsoftbus_feature_conn_br) {
+  if (dsoftbus_feature_conn_br && support_jsapi) {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common.c" ]
   } else {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common_virtual.c" ]
@@ -133,7 +133,7 @@ ohos_unittest("GeneralConnectionConstraintTest") {
     "$dsoftbus_root_path/tests/core/connection/general/unittest/softbus_conn_general_connection_constraint_test.cpp",
   ]
 
-  if (dsoftbus_feature_conn_br) {
+  if (dsoftbus_feature_conn_br && support_jsapi) {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common.c" ]
   } else {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common_virtual.c" ]
@@ -186,7 +186,7 @@ ohos_unittest("GeneralNegotiationTest") {
     "$dsoftbus_root_path/tests/core/connection/general/unittest/softbus_conn_general_negotiation_test.cpp",
   ]
 
-  if (dsoftbus_feature_conn_br) {
+  if (dsoftbus_feature_conn_br && support_jsapi) {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common.c" ]
   } else {
     sources += [ "$dsoftbus_root_path/adapter/common/net/bluetooth/common/softbus_adapter_bt_common_virtual.c" ]
diff --git a/tests/core/frame/unittest/BUILD.gn b/tests/core/frame/unittest/BUILD.gn
index b09e90f64..5d964c30c 100644
--- a/tests/core/frame/unittest/BUILD.gn
+++ b/tests/core/frame/unittest/BUILD.gn
@@ -185,6 +185,9 @@ ohos_unittest("SoftbusClientInfoManagerTest") {
 ohos_unittest("SoftbusGEnhanceTest") {
   module_out_path = module_output_path
   sources = [ "softbus_g_enhance_test.cpp" ]
+  if (support_jsapi) {
+    defines = [ "SUPPORT_JSAPI" ]
+  }
 
   include_dirs = [
     "$dsoftbus_dfx_path/interface/include/form",
@@ -230,7 +233,7 @@ group("unittest") {
     ":SoftbusGEnhanceTest",
     ":SoftbusServerTest",
   ]
-  if (dsoftbus_access_token_feature) {
+  if (dsoftbus_access_token_feature && support_jsapi) {
     deps += [ ":SoftbusServerStubTest" ]
   }
 }
diff --git a/tests/core/frame/unittest/softbus_g_enhance_test.cpp b/tests/core/frame/unittest/softbus_g_enhance_test.cpp
index fcd0eaa32..b8c06cd13 100644
--- a/tests/core/frame/unittest/softbus_g_enhance_test.cpp
+++ b/tests/core/frame/unittest/softbus_g_enhance_test.cpp
@@ -934,6 +934,7 @@ void DiscShareNfcDeinitStub(void)
  * @tc.level: Level1
  * @tc.require:
  */
+#ifdef SUPPORT_JSAPI
 HWTEST_F(SoftbusGEnhanceTest, SoftbusGEnhanceTest029, TestSize.Level1)
 {
     DiscEnhanceFuncList *pfnDiscEnhanceFuncList = DiscEnhanceFuncListGet();
@@ -943,21 +944,21 @@ HWTEST_F(SoftbusGEnhanceTest, SoftbusGEnhanceTest029, TestSize.Level1)
     pfnDiscEnhanceFuncList->schedulerSetBroadcastParam = SchedulerSetBroadcastParamStub;
     ret = SchedulerSetBroadcastParamPacked(0, nullptr);
     EXPECT_EQ(ret, SOFTBUS_OK);
- 
+
     pfnDiscEnhanceFuncList->schedulerInitBroadcast = nullptr;
     ret = SchedulerInitBroadcastPacked();
     EXPECT_EQ(ret, SOFTBUS_OK);
     pfnDiscEnhanceFuncList->schedulerInitBroadcast = InitBroadcastMgrStub;
     ret = SchedulerInitBroadcastPacked();
     EXPECT_EQ(ret, SOFTBUS_OK);
- 
+
     pfnDiscEnhanceFuncList->schedulerDeinitBroadcast = nullptr;
     ret = SchedulerDeinitBroadcastPacked();
     EXPECT_EQ(ret, SOFTBUS_OK);
     pfnDiscEnhanceFuncList->schedulerDeinitBroadcast = DeInitBroadcastMgrStub;
     ret = SchedulerDeinitBroadcastPacked();
     EXPECT_EQ(ret, SOFTBUS_OK);
- 
+
     pfnDiscEnhanceFuncList->discCoapProcessDeviceInfo = nullptr;
     DiscInnerCallback discCb = {0};
     ret = DiscCoapProcessDeviceInfoPacked(nullptr, nullptr, discCb);
@@ -988,6 +989,7 @@ HWTEST_F(SoftbusGEnhanceTest, SoftbusGEnhanceTest029, TestSize.Level1)
     ret = DiscFillBtypePacked(0, 0, nullptr);
     EXPECT_EQ(ret, SOFTBUS_OK);
 }
+#endif
 
 /**
  * @tc.name: SoftbusGEnhanceTest030
diff --git a/tests/core/transmission/session/BUILD.gn b/tests/core/transmission/session/BUILD.gn
index 5037b9248..0c8cf97a0 100644
--- a/tests/core/transmission/session/BUILD.gn
+++ b/tests/core/transmission/session/BUILD.gn
@@ -84,34 +84,38 @@ ohos_unittest("TransSessionServiceTest") {
   ]
 }
 
-ohos_unittest("TransScenarioManagerTest") {
-  module_out_path = module_output_path
-  sources = [ "softbus_scenario_manager_test.cpp" ]
+if (support_jsapi) {
+  ohos_unittest("TransScenarioManagerTest") {
+    module_out_path = module_output_path
+    sources = [ "softbus_scenario_manager_test.cpp" ]
 
-  include_dirs = [
-    "$dsoftbus_core_path/common/include",
-    "$dsoftbus_core_path/transmission/session/include",
-    "$dsoftbus_core_path/transmission/session/src",
-  ]
+    include_dirs = [
+      "$dsoftbus_core_path/common/include",
+      "$dsoftbus_core_path/transmission/session/include",
+      "$dsoftbus_core_path/transmission/session/src",
+    ]
 
-  deps = [
-    "$dsoftbus_core_path/common:softbus_utils",
-    "$dsoftbus_core_path/frame:softbus_server",
-    "$dsoftbus_root_path/adapter:softbus_adapter",
-  ]
+    deps = [
+      "$dsoftbus_core_path/common:softbus_utils",
+      "$dsoftbus_core_path/frame:softbus_server",
+      "$dsoftbus_root_path/adapter:softbus_adapter",
+    ]
 
-  external_deps = [
-    "c_utils:utils",
-    "hilog:libhilog",
-    "wifi:wifi_sdk",
-  ]
+    external_deps = [
+      "c_utils:utils",
+      "hilog:libhilog",
+      "wifi:wifi_sdk",
+    ]
+  }
 }
 
 group("unittest") {
   testonly = true
   deps = [
-    ":TransScenarioManagerTest",
     ":TransSessionManagerTest",
     ":TransSessionServiceTest",
   ]
+  if (support_jsapi) {
+    deps += [ ":TransScenarioManagerTest" ]
+  }
 }
